controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -909,6 +907,31 @@ func (pp *portPublisher) endpointSliceToAddresses(es *discovery.EndpointSlice) A
 			}
 		}
 
+		if endpoint.TargetRef.Kind == endpointTargetRefExternalWorkload {
+			for _, IPAddr := range endpoint.Addresses {
+				address, id, err := pp.newExtRefAddress(resolvedPort, IPAddr, endpoint.TargetRef.Name, endpoint.TargetRef.Namespace)
+				if err != nil {
+					pp.log.Errorf("Unable to create new address:%v", err)
+					continue
+				}
+
+				err = SetToServerProtocolExternalWorkload(pp.k8sAPI, &address, resolvedPort)
+				if err != nil {
+					pp.log.Errorf("failed to set address OpaqueProtocol: %s", err)
+					continue
+				}
+				address.Zone = endpoint.Zone
```

Comment:
How hard is it to create a new string here? Looks like a benign line but we end up with a reference to a string on the endpoint. I'm not sure that we'd ever mutate the inner value, or that we'd be keeping the endpoint data from being gc'd, but just in case should we create a new string altogether? wdyt?